assets/helpers/paymentIntegrations/readerRevenueApis.js

```diff
@@ -37,11 +37,22 @@ type RegularContribution = {|
   billingPeriod: BillingPeriod,
 |};
 
-type PayPalDetails = {| baid: string |};
+// TODO: can we do away with these types and use the PaymentAuthorisation here?
+// and thus do away with getPaymentFields and paymentDetailsFromAuthorisation
+// (would probably require backend renaming)
+export type PayPalDetails = {| baid: string |};
 
-type StripeDetails = {| stripeToken: string |};
+export type StripeDetails = {| stripeToken: string |};
 
-type PaymentDetails = PayPalDetails | StripeDetails;
+export type DirectDebitDetails = {|
+  accountHolderName: string,
+  sortCode: string,
+  accountNumber: string,
+|};
+
+// TODO: rename this type and its constituent types since the below structure is a bit baffling
+// PaymentFields: {contributionType, fields: {...other stuff, paymentFields: PaymentDetails}}
```

Comment:
It is indeed how compilers tag values in a sealed trait.

Reply:
I actually mean that the top-level object is of type `PaymentFields`, which contains `fields.paymentFields` which is of type `PaymentDetails`

Reply:
lol, I did not see that.